src/main/java/hudson/plugins/git/GitPublisher.java

```diff
@@ -185,7 +186,7 @@ public boolean perform(AbstractBuild<?, ?> build,
         SCM scm = build.getProject().getScm();
 
         if (!(scm instanceof GitSCM)) {
-            return false;
+            return true; // just skip this publisher
```

Comment:
Questionable. If you configured the publisher, presumably you meant it to be used. Better to fail with an explanation than silently do nothing.